pkg/controller/storagecluster/update.go

```diff
@@ -529,6 +561,18 @@ func (c *Controller) isPodUpdated(
 		logrus.Warnf("Could not check the diff between current pod and latest spec. %v", err)
 		return false
 	}
+
+	// Pod not matched will be deleted then will be annotated upon restart.
+	// Updated pod will be annotated directly with known and custom annotations without restart,
+	// this case pod hash needs to be updated as well to avoid inconsistency
+	if matched {
```

Comment:
move to outside of isPodUpdated function, the name isPodUpdated indicate checking if pod is updated instead of updating the pod.

Reply:
able to move it to `getAllStorageClusterPods` same it's not a good place to make an update. 
update stc spec annotation will change the stc hash, and we don't want a pod restart, then in logic putting it in `isPodUpdated` after checking the condition pod's hash is equal to stc hash, but before returning the pod is updated should be the best place IMO.

Reply:
hash should not change with custom annotations right? i am not sure if i get it. in rollingUpdate function, it returned newPods here, we can update annotations here.

_, oldPods := c.getAllStorageClusterPods(cluster, nodeToStoragePods, hash)

Reply:
return two flags for caller to handle in isPodUpdated, one flag for restarting, and one for update without restarting. we can rename isPodUpdated to podNeedsUpdate()